src/lib/SaferNames/Inference.hs

```diff
@@ -640,7 +640,19 @@ checkOrInferRho (WithSrcE pos expr) reqTy = do
     value' <- zonk =<< (checkRho value $ VariantTy $ Ext NoLabeledItems $ Just row)
     prev <- mapM (\() -> freshType TyKind) labels
     matchRequirement =<< emitOp (VariantLift prev value')
-  UIntLit  x  -> matchRequirement $ Con $ Lit  $ Int32Lit $ fromIntegral x
+  UIntLit x  -> do
+    lookupSourceMap MethodNameRep "fromInteger" >>= \case
```

Comment:
What's wrong with protolude? I also have follow-up patches that add e.g. the `Ix` typeclass over there and I'm pretty sure that one will have to be more special than `FromInteger`. I would imagine that the same code should just work, why delete it?

Reply:
I also find it a bit questionable to make it so that the language semantics change once a surface-level definition is introduced.

Reply:
I don't have anything against the protolude in principle. This was just quicker to implement. We can add a protolude if we need it for `Ix` (though I suspect we might be able to use this approach too).